otj-kafka/src/main/java/com/opentable/kafka/logging/GenerateHeaders.java

```diff
@@ -0,0 +1,19 @@
+package com.opentable.kafka.logging;
+
+import java.util.Arrays;
+
+/**
+ * Controls the generation of headers.
+ * Default = ALL
+ * Rationale is that for small messages, there's actually significant overhead here.
+ */
+public enum GenerateHeaders {
+    ALL,  // Propagate headers including TRACE, conserved headers, and informational headers
```

Comment:
Allows fine grained control. Mostly for debugging - we don't formally expose these so to set them requires addProperty()